src/controllers/OAuthController.ts

```diff
@@ -0,0 +1,28 @@
+import { type Request, type Response } from 'express';
+import { OAuthProvider } from '../providers/OAuthProvider';
+
+export class OAuthController {
+  auth(_req: Request, res: Response): void {
+    const state = 'state';
+    const oauthProvider = new OAuthProvider();
+    const redirectURI = oauthProvider.getRedirectUri(state);
+    res.cookie('auth_state', state);
+    res.status(200).redirect(redirectURI);
+  }
+
+  async callback(req: Request, res: Response): Promise<Response> {
+    const { code, state } = req.query;
+    const storedState = req.cookies !== null ? req.cookies['auth_state' as string] : null;
```

Comment:
Creio que validação ainda mais efeitva seria comparando se o `req. cookies` for **true**. Assim, se ele vinher como **undefined** ou uma string vazia, o `storedState` irá ser setado como **null**.

Reply:
Tinha feito assim, so que tive um problema com o eslint e não queria adicionar a regra, isso foi um ajuste de preguiçoso.

Reply:
O erro é esse https://typescript-eslint.io/rules/strict-boolean-expressions. Se puder da uma olhada e um feedback.